code/modules/modular_computers/hardware/card_slot.dm

```diff
@@ -6,14 +6,29 @@
 	w_class = WEIGHT_CLASS_TINY
 	device_type = MC_CARD
 
-	var/obj/item/card/id/stored_card = null
-	var/obj/item/card/id/stored_card2 = null
+	var/obj/item/card/id/stored_card
+	var/obj/item/card/id/stored_card2
 
 /obj/item/computer_hardware/card_slot/Exited(atom/A, atom/newloc)
-	if(A == stored_card)
-		try_eject(1, null, TRUE)
-	if(A == stored_card2)
-		try_eject(2, null, TRUE)
+	if(!(A == stored_card || A == stored_card2))
+		return ..()
+	if(!holder)
+		return ..()
+
+	if(holder.active_program)
+		holder.active_program.event_idremoved(FALSE)
+	for(var/datum/computer_file/program/computer_program as anything in holder.idle_threads)
+		computer_program.event_idremoved(TRUE)
```

Comment:
can you use named variables? meaning (background = TRUE), makes the code more readable without someone having to switch to a completely different proc